packages/gatsby-cli/src/init-starter.js

```diff
@@ -27,15 +28,56 @@ const shouldUseYarn = () => {
   }
 }
 
+const checkYarnVersion = () => {
+  let hasMinYarnPnp = false
+  let yarnVersion = null
+  try {
+    yarnVersion = execSync(`yarnpkg --version`)
+      .toString()
+      .trim()
+    hasMinYarnPnp = semver.gte(yarnVersion, `1.12.0`)
+  } catch (err) {
+    // ignore
+  }
+  return {
+    hasMinYarnPnp: hasMinYarnPnp,
+    yarnVersion: yarnVersion,
+  }
+}
+
 // Executes `npm install` or `yarn install` in rootPath.
-const install = async rootPath => {
+const install = async ({ rootPath, usePnp }: Options) => {
   const prevDir = process.cwd()
 
   report.info(`Installing packages...`)
   process.chdir(rootPath)
 
   try {
-    let cmd = shouldUseYarn() ? spawn(`yarnpkg`) : spawn(`npm install`)
+    let cmd
+    if (shouldUseYarn()) {
+      const yarnInfo = checkYarnVersion()
+      const args = []
+      if (usePnp) {
+        if (!yarnInfo.hasMinYarnPnp) {
+          if (yarnInfo.yarnVersion) {
+            report.warn(
+              `You are using Yarn ${
+                yarnInfo.yarnVersion
+              } together with the --use-pnp flag, but Plug'n'Play is only supported starting from the 1.12 release.\n\n` +
+                `Please update to Yarn 1.12 or higher for a better, fully supported experience.\n`
```

Comment:
will need to update this with the version that comes out of https://github.com/yarnpkg/yarn/pull/6682